insert_string.c

```diff
@@ -10,7 +10,7 @@
 
 #define HASH_SLIDE           16
 
-#define HASH_CALC(h, val)    h = ((val * 2654435761U) >> HASH_SLIDE);
+#define HASH_CALC(h, val)    h = ((val * 66521U + 66521U) >> HASH_SLIDE);
```

Comment:
![image](https://github.com/zlib-ng/zlib-ng/assets/1364432/bf5219bc-bb9a-4df7-a75e-e480563af8a3)

Reply:
https://chromium.googlesource.com/chromium/src/third_party/zlib/+/91a295c72eeba2b16e51350c0ca8de8ed160f6fc

Reply:
Ah, there we have it, `ANZAC++ hash`.

I have no idea whether it is any better or worse than our current one, but so far I don't see any compelling reason to change.
On x86-64, it compiles to one extra `lea` instruction, so hardly a big problem performance-wise.
But I kind of suspect it just changes around what test data compresses better, instead of actually moving the average compression ratio.

Reply:
The only sure way I know of to improve compression by reducing hash collisions is to increase the size of the hashtable, but that gives diminishing returns for sure. As far as I can remember I think we already use one that is twice the size of the one upstream zlib uses at max size.

Reply:
More background and suggested further changes to ANZAC++ here:
https://groups.google.com/a/chromium.org/g/chromium-reviews/c/g13xpkLOt10

Reply:
According to the two benchmark spreadsheets, they don't seem to have evaluated Knuth at all.

Reply:
Here is a link to the Gerrit page, which may be easier to read..
https://chromium-review.googlesource.com/c/chromium/src/+/4880264